libsolidity/analysis/TypeChecker.cpp

```diff
@@ -433,6 +433,8 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 		m_errorReporter.typeError(_function.location(), "Constructor must be implemented if declared.");
 	else if (isLibraryFunction)
 		m_errorReporter.typeError(_function.location(), "Library functions must be implemented if declared.");
+	else if (!_function.virtualSemantics())
+		m_errorReporter.typeError(_function.location(), "Functions without implementation must be marked virtual.");
```

Comment:
Why is it the case here that the function doesn't have an implementation?

Reply:
It's among the ``else`` cases of ``if (function.isImplemented())`` line 430.